InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Validations.Modify.cs

```diff
@@ -0,0 +1,391 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE TO CONNECT THE WORLD
+// ---------------------------------------------------------------
+
+using FluentAssertions;
+using Force.DeepCloner;
+using InternTrack.Core.Api.Models.Interns;
+using InternTrack.Core.Api.Models.Interns.Exceptions;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace InternTrack.Core.Api.Tests.Unit.Services.Foundations.Interns
+{
+    public partial class InternServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowValidationExceptionOnModifyWhenInternIsNullAndLogItAsync()
+        {
+            // given
+            Intern nullIntern = null;
+
+            var nullInternException =
+                new NullInternException();
+
+            var expectedInternValidationException =
+                new InternValidationException(nullInternException);
+
+            // when
+            ValueTask<Intern> modifyInternTask =
+                this.internService.ModifyInternAsync(nullIntern);
+
+            InternValidationException actualInternValidationException =
+                await Assert.ThrowsAsync<InternValidationException>(
+                    modifyInternTask.AsTask);
+
+            // then
+            actualInternValidationException.Should().BeEquivalentTo(
+                expectedInternValidationException);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameValidationExceptionAs(
+                    expectedInternValidationException))),
+                        Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.InsertInternAsync(It.IsAny<Intern>()),
+                    Times.Never);
+
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData(" ")]
+        public async Task ShouldThrowValidationExceptionOnModifyWhenInternIsInvalidAndLogItAsync(string invalidText)
+        {
+            // given
+            var invalidIntern = new Intern
+            {
+                Id = Guid.Empty,
+                FirstName = invalidText,
+                MiddleName = invalidText,
+                LastName = invalidText,
+                Email = invalidText,
+                PhoneNumber = invalidText,
+                Status = invalidText,
+                UpdatedDate = default,
+                CreatedDate = default,
+                JoinDate = default,
+                UpdatedBy = Guid.Empty,
+                CreatedBy = Guid.Empty
```

Comment:
If you leave the property alone without an assignment if it has one potential invalid value type then just leave it alone without an assignment and it should pick up the invalid value automatically